### PR TITLE
Contributions

### PR DESCRIPTION
**Pull request by Maltesius**

### CODE REVIEW

File: report/src/sections/01-introduction.tex [hunk 1]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
 \subsection{Contributions}\label{subsec:contributions}
 \todo[inline]{Casper says: These section (\ref{subsec:contributions}, \ref{subsec:related-work}, \ref{subsec:article-overview}) read like notes, I assume they are.}
 
-Use Halder concretely for abstract interpretation
+%Use Halder concretely for abstract interpretation:
+This paper builds on the work of abstract interpretation of database queries by Cortesi and Halder.
+Concretely, we use their work to analyze a database schema concerning a program using an alteration of the syntax they describe.
```
Comment (Daniel-Runge) [inline report/src/sections/01-introduction.tex:56]: This is already in the related works. Probably delete this paragraph, or rewrite it to refer to related works.

Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:56]: "Alternation" refers to the process of repeatedly changing from one state, condition, action, or place to another and back again.

File: report/src/sections/01-introduction.tex [hunk 2]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
 
-This paper builds on work of abstract interpretation of database queries by Cortesi and Halder.
-Concretely, we use their work for analysing a database schema in relation to a program using the syntax they describe.
+%Improve on the levels of abstraction, i.e., add not just abstract values but also abstract tuples:
 
+Also, by building upon their work, we extend the level of abstraction used in the analysis.
+Where they only abstracted over values, this paper also abstracts over tuples in the database, ultimately guaranteeing a finite analysis.
```
Comment (Daniel-Runge) [inline report/src/sections/01-introduction.tex:61]: Rewrite this paragraph after correcting the one above.

File: report/src/sections/01-introduction.tex [hunk 3]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+This is thoroughly covered in \autoref{sec:theory}.
 
-Improve on the levels of abstraction, i.e., add not just abstract values but also abstract tuples
+%Describe an implementable algorithm for performing static analysis targeting specific properties, given a program and database:
 
-Describe an implementable algorithm for performing static analysis targeting specific properties, given a program and database.
+Also included in the contribution is an implementable algorithm, which can be used for performing static analysis.
+This algorithm targets specific properties when given a program and its associated database.
```
Comment (Daniel-Runge) [inline report/src/sections/01-introduction.tex:67]: Do we actually have this algorithm?

**Review by Daniel-Runge — CHANGES_REQUESTED** [review]

File: report/src/sections/01-introduction.tex (hunk 2 above)
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:61]: We abstract over the multiplicity of the tuples and the values also, they only abstract over the tuple and no the multiplicity.

File: report/src/sections/01-introduction.tex (hunk 3 above)
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:67]: no.

File: report/src/sections/01-introduction.tex [hunk 4]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+The allowed properties can be seen in the following \autoref{subsubsec:properties}.
 
-Prove soundness for the analysis.
+%Prove soundness for the analysis:
+To ensure that our abstractions are correct, we informally prove their soundness.
+By doing this, we can always be sure that our abstractions are an overestimation of the concrete values and tuples.
+This ensures that an abstract value always represents a concrete value.
```
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:73]: This is confusing the abstractions with the analysis. We prove that the analysis is sound, that is to say we prove that the analysis assignment that the analysis produces is an overestimation (see preliminaries).

File: report/src/sections/01-introduction.tex [hunk 5]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+The soundness proof can be found in \autoref{subsec:soundness}.
 
+%Describe the properties we wish to check using LTL-like encodings, i.e., phrases like \emph{always, eventually,} and \emph{never}, to describe undesirable database states.\todo{Casper says: As of now we are only able to check invariants.}
+\subsubsection{Properties} \label{subsubsec:properties}
```
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:77]: Why is this a sub sub section?

File: report/src/sections/01-introduction.tex [hunk 6]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+While we have not provided a specific method for checking general properties, we have outlined a systematic approach for verifying invariants.
+Invariants are a particular type of property that should always be held during the program's execution.
 
-Describe the properties we wish to check using LTL-like encodings, i.e., phrases like \emph{always, eventually,} and \emph{never}, to describe undesirable database states.\todo{Casper says: As of now we are only able to check invariants.}
-Additionally, we use LTL as described in X, Y, and Z to describe properties to check against.
+This way, we can check at each step of the program to see if the invariant holds, and if it does not, we can stop the analysis and report that the program is incorrect.
```
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:81]: We should all agree this is the case... Remember from sMV you make the analyisis and then you check if the property holds.

File: report/src/sections/01-introduction.tex [hunk 7]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+
+We can check for invariants in every state of the program.
+We do this in the program graph, where each node represents a state of the program.
+We can check if the invariant holds in each node.
+This way, we can systematically verify the program's correctness throughout its execution.
+
+Describing an invariant is done using $\forall$, $\exists$, $\land$, $\lor$, and $\neg$.
```
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:88]: We could be clever here and say that invariants can be check with the predicate syntax we have already describe.

File: report/src/sections/01-introduction.tex [hunk 8]
```diff
@@ -51,21 +51,49 @@ \subsection{Non-terminating Analysis}\label{subsec:non-terminating-analysis}
...
+
+An example of an invariant is:
+\begin{equation}
+    \forall x \in \mathbb{N} : x \geq 0
+\end{equation}
+
+This invariant states that for all $x$ in the set of natural numbers $\mathbb{N}$, $x$ must be greater than or equal to 0.
+The invariant must hold in every state of the program where x is considered.
```
Comment (CasperStaahl) [inline report/src/sections/01-introduction.tex:96]: Weird example.

**Review by CasperStaahl — DISMISSED** [review]

**Review by Daniel-Runge — APPROVED** [review]

**Pull request merged**